homeassistant/components/sunsynk/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""The Sunsynk integration."""
+from __future__ import annotations
+
+from sunsynk.client import SunsynkClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DATA_INVERTER_SN, DOMAIN, SUNSYNK_COORDINATOR
+from .coordinator import SunsynkCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Sunsynk from a config entry."""
+    inverter_sn = entry.data[DATA_INVERTER_SN]
+
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    inverter_sn = entry.data[DATA_INVERTER_SN]
+
+    api = await SunsynkClient.create(username, password)
+
+    sunsunk_coordinator = SunsynkCoordinator(hass, api, inverter_sn)
```

Comment:
any reason we're not doing `await sunsunk_coordinator.async_config_entry_first_refresh()`

Reply:
No there isn't. I've updated this to call `await sunsunk_coordinator.async_config_entry_first_refresh()`